Svc/BufferRepeater/BufferRepeater.cpp

```diff
@@ -0,0 +1,81 @@
+// ======================================================================
+// \title  BufferRepeater.cpp
+// \author lestarch
+// \brief  cpp file for GenericRepeater component implementation class
+//
+// \copyright
+// Copyright 2009-2015, by the California Institute of Technology.
+// ALL RIGHTS RESERVED.  United States Government Sponsorship
+// acknowledged.
+//
+// ======================================================================
+
+#include <FpConfig.hpp>
+#include <Svc/BufferRepeater/BufferRepeater.hpp>
+
+namespace Svc {
+
+// ----------------------------------------------------------------------
+// Construction, initialization, and destruction
+// ----------------------------------------------------------------------
+
+BufferRepeater ::BufferRepeater(const char* const compName)
+    : BufferRepeaterComponentBase(compName),
+      m_allocation_failure_response(BufferRepeater::NUM_BUFFER_REPEATER_FAILURE_OPTIONS) {}
+
+void BufferRepeater ::init(const NATIVE_INT_TYPE instance) {
+    BufferRepeaterComponentBase::init(instance);
+}
+
+BufferRepeater ::~BufferRepeater() {}
+
+void BufferRepeater ::configure(BufferRepeater::BufferRepeaterFailureOption allocation_failure_response) {
+    this->m_allocation_failure_response = allocation_failure_response;
+}
+
+bool BufferRepeater ::check_allocation(FwIndexType index,
```

Comment:
Should we assert that index < number of ports?

Reply:
yup